fe/fe-core/src/main/java/com/starrocks/scheduler/mv/IMTCreator.java

```diff
@@ -0,0 +1,231 @@
+            distributionInfo.setBucketNum(numBucket);
+        }
+
+        // Refresh
+        MaterializedView.MvRefreshScheme mvRefreshScheme = new MaterializedView.MvRefreshScheme();
+        mvRefreshScheme.setType(MaterializedView.RefreshType.INCREMENTAL);
+
+        String mvName = stmt.getTableName().getTbl();
+        return new MaterializedView(mvId, dbId, mvName, columns, stmt.getKeysType(), partitionInfo,
+                distributionInfo, mvRefreshScheme);
+    }
+
+    public static void createIMT(CreateMaterializedViewStatement stmt, MaterializedView view) throws DdlException {
+        ExecPlan maintenancePlan = stmt.getMaintenancePlan();
+        OptExpression optExpr = maintenancePlan.getPhysicalPlan();
+        List<CreateTableStmt> createTables = IMTCreatorVisitor.createIMTForOperator(stmt, optExpr, view);
+
+        for (CreateTableStmt create : createTables) {
+            LOG.info("creating IMT {} for MV {}", create.getTableName(), view.getName());
+            try {
+                GlobalStateMgr.getCurrentState().createTable(create);
+            } catch (DdlException e) {
+                // TODO(murphy) cleanup created IMT, or make it atomic
+                LOG.info("create IMT {} failed due to ", create.getTableName(), e);
+                throw e;
+            }
+        }
+    }
+
+    static class IMTCreatorVisitor extends OptExpressionVisitor<Void, Void> {
+        private CreateMaterializedViewStatement stmt;
+        private MaterializedView view;
+        private List<CreateTableStmt> result = new ArrayList<>();
+
+        public static List<CreateTableStmt> createIMTForOperator(CreateMaterializedViewStatement stmt,
+                                                                 OptExpression optExpr,
+                                                                 MaterializedView view) {
+            IMTCreatorVisitor visitor = new IMTCreatorVisitor();
+            visitor.stmt = stmt;
+            visitor.view = view;
+            visitor.visit(visitor, optExpr);
+            return visitor.result;
+        }
+
+        private void visit(IMTCreatorVisitor visitor, OptExpression optExpr) {
+            for (OptExpression child : optExpr.getInputs()) {
+                visitor.visit(child, null);
+            }
+            visitor.visit(optExpr, null);
+        }
+
+        @Override
+        public Void visit(OptExpression optExpr, Void ctx) {
+            return null;
+        }
+
+        @Override
+        public Void visitPhysicalStreamJoin(OptExpression optExpr, Void ctx) {
+            // Join does not need any IMT
+            return null;
+        }
+
+        /**
+         * StreamAgg needs two IMT
+         * 1. State Table: store the state of all aggregation functions for non-retractable ModifyOp
+         * 2. Detail Table: for retractable ModifyOp
+         * <p>
+         * TODO(murphy) create detail table
+         */
```

Comment:
Now StreamAgg may depend 3 kinds of state tables:
1. ResultStateTable, keeps the result of StreamAgg node which can be used to generate RetractMessage and intermediate state for some kinds of AggFuncs, eg min/max/sum/count;
2. IntermediateStateTable, keeps the intermediate state of generate AggFuncs, such as avg/approx_count_distinct/retract_min;
3. DetailStateTable, keeps the detail data of each detail-kind AggFuncs, such as min/max.

Reply:
Yes, you could specify the stat in later PR.